src/ImageProcessor.Web.Plugins.PostProcessor/PostProcessor.cs

```diff
@@ -99,16 +98,17 @@ public static async Task<MemoryStream> PostProcessImageAsync(HttpContext context
                         }
                     }
 
-                    // Save result
+                    // Use source file if no destination file is created
                     if (!File.Exists(destinationFile))
                     {
-                        File.Copy(sourceFile, destinationFile, true);
+                        destinationFile = sourceFile;
                     }
 
+                    // Save result
                     var result = new PostProcessingResultEventArgs(destinationFile, length);
```

Comment:
Any reason why we're using the additional `PostProcessingResultEventArgs` class, instead of getting the result file size and check whether it's smaller than the source (but not zero) right here?

Reply:
I've removed the use of `PostProcessingResultEventArgs` and obsoleted it (as it was publicly exposed). We could also just remove the class completely, as it's probably not used by external code anyway.

Reply:
It's now completely removed 👍